experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -377,94 +403,42 @@ function rebaseMarkIgnoreChild<TNodeChange>(
 	moveEffects: MoveEffectTable<TNodeChange>,
 	nodeExistenceState: NodeExistenceState,
 ): Mark<TNodeChange> {
-	let rebasedMark = currMark;
-	if (markEmptiesCells(baseMark)) {
-		assert(isDetach(baseMark), 0x70b /* Only detach marks should empty cells */);
+	let rebasedMark;
+	if (isDetach(baseMark)) {
+		if (baseMark.cellId !== undefined) {
+			// Detaches on empty cells have an implicit revive effect.
+			delete currMark.cellId;
+		}
+		assert(
+			!isNewAttach(currMark),
+			0x69d /* A new attach should not be rebased over its cell being emptied */,
+		);
 		const baseCellId = getDetachOutputId(baseMark, baseRevision, metadata);
 
-		// TODO: Should also check if this is a transient move source
 		if (isMoveSource(baseMark)) {
 			assert(isMoveMark(baseMark), 0x6f0 /* Only move marks have move IDs */);
-			if (markFollowsMoves(rebasedMark)) {
+			const { remains, follows } = separateEffectsForMove(currMark);
+			if (follows !== undefined) {
 				sendMarkToDest(
-					rebasedMark,
+					follows,
 					moveEffects,
 					getEndpoint(baseMark, baseRevision),
 					baseMark.count,
 				);
-				return { count: baseMark.count, cellId: cloneCellId(baseCellId) };
-			}
-
-			const modify = rebasedMark.changes;
-			if (modify !== undefined) {
-				rebasedMark = withNodeChange(rebasedMark, undefined);
-				const nestedChange: CellMark<NoopMark, TNodeChange> = {
-					count: 1,
-					changes: modify,
-				};
-				sendMarkToDest(
-					nestedChange,
-					moveEffects,
-					getEndpoint(baseMark, baseRevision),
-					baseMark.count,
-				);
-			}
-		}
-
-		assert(
-			!isNewAttach(rebasedMark),
-			0x69d /* A new attach should not be rebased over its cell being emptied */,
-		);
-
-		if (isMoveMark(rebasedMark)) {
-			if (rebasedMark.type === "MoveOut" || rebasedMark.type === "ReturnFrom") {
-				setPairedMarkStatus(
-					moveEffects,
-					CrossFieldTarget.Destination,
-					getEndpoint(rebasedMark, undefined),
-					rebasedMark.count,
-					PairedMarkUpdate.Deactivated,
-				);
-			} else if (isReattach(rebasedMark)) {
-				setPairedMarkStatus(
-					moveEffects,
-					CrossFieldTarget.Source,
-					getEndpoint(rebasedMark, undefined),
-					rebasedMark.count,
-					PairedMarkUpdate.Reactivated,
-				);
 			}
+			rebasedMark = remains ?? { count: baseMark.count };
```

Comment:
Aren't the remaining effects targeting empty cells? So the noop should have a cell ID?

Reply:
Same with `remains`.